geaflow/geaflow-dsl/geaflow-dsl-connector/geaflow-dsl-connector-kafka/src/main/java/com/antgroup/geaflow/dsl/connector/kafka/KafkaTableSource.java

```diff
@@ -121,7 +124,13 @@ public List<Partition> listPartitions() {
 
     @Override
     public <IN> TableDeserializer<IN> getDeserializer(Configuration conf) {
-        return (TableDeserializer<IN>) new TextDeserializer();
+        this.connectorFormat = conf.getString(ConnectorConfigKeys.GEAFLOW_DSL_CONNECTOR_FORMAT,
+                (String) ConnectorConfigKeys.GEAFLOW_DSL_CONNECTOR_FORMAT.getDefaultValue());
+        if (this.connectorFormat.equals(ConnectorConstants.CONNECTOR_FORMAT_JSON)) {
```

Comment:
It's recommended to use a factory class, so that components can be modified or added/removed in the future more easily.

Reply:
@Leomrlin sure, another question is currently some table sources directly return TableDeserializer object, for example `HiveTableSource` return `new RowTableDeserializer();`. If I introduce a factory class for TableDeserializer, do you think it's appropriate for all the TableSource to use factory class?

Reply:
> @Leomrlin sure, another question is currently some table sources directly return TableDeserializer object, for example `HiveTableSource` return `new RowTableDeserializer();`. If I introduce a factory class for TableDeserializer, do you think it's appropriate for all the TableSource to use factory class?

It would be ideal if all types of Deserializers could be unified into a factory class. Upon reviewing, I have noted that there are currently two basic implementation classes: RowTableDeserializer and TextDeserializer. In the future, there might be additional implementation classes. If they are exclusive to a plugin like Paimon, they can be included within the plugin package. I think that JsonDeserializer could serve as a basic implementation class and be incorporated into the factory class.